artichoke-backend/src/extn/core/kernel/mod.rs

```
@@ -108,6 +120,57 @@ impl Kernel {
+        let interp = unwrap_interpreter!(mrb);
+
+        match integer::Args::extract(&interp) {
+            Ok(args) => match integer::method(&interp, &args) {
+                Ok(v) => v.inner(),
+                Err(integer::Error::InvalidRadix(v, raise_exception)) => {
+                    if raise_exception {
+                        ArgumentError::raisef(interp, "invalid radix", vec![v])
+                    } else {
+                        interp.convert(None::<Value>).inner()
+                    }
+                }
+                Err(integer::Error::InvalidValue(v, raise_exception)) => {
+                    if raise_exception {
+                        ArgumentError::raisef(
+                            interp,
+                            "invalid value for Integer(): \"%S\"",
+                            vec![v],
+                        )
+                    } else {
+                        interp.convert(None::<Value>).inner()
+                    }
+                }
+                Err(integer::Error::ContainsNullByte(raise_exception)) => {
+                    if raise_exception {
+                        ArgumentError::raise(interp, "string contains null byte")
+                    } else {
+                        interp.convert(None::<Value>).inner()
+                    }
+                }
+                _ => RuntimeError::raise(interp, "fatal Kernel#Integer error"),
+            },
+            Err(integer::Error::NoImplicitConversionToString(class_name, raise_exception)) => {
+                if raise_exception {
+                    TypeError::raisef(interp, "can't convert %S into String", vec![class_name])
+                } else {
+                    interp.convert(None::<Value>).inner()
+                }
+            }
+            Err(integer::Error::BaseSpecifiedForNonString(raise_exception)) => {
+                if raise_exception {
```

if we were to do the error swallowing in Rust, we could do it more concisely with

```rust
match result {
    Ok(value) => value.inner(),
    Err(integer::Error::NoImplicitConversionToString(.., raise_exception)) | Err(integer::Error::BaseSpecifiedForNonString(raise_exception)) | ... if !raise_exception => interp.convert(None::<Value>).inner(),
    ...
}
```